# Hospitalization Hypothesis Testing

Apollo Hospitals was established in 1983, renowned as the architect of modern healthcare in India. As the nation's first corporate hospital, Apollo Hospitals is acclaimed for pioneering the private healthcare revolution in the country.

As a data scientist, the ultimate goal is to tease out meaningful and actionable insights from Patient-level collected data. You can help Apollo hospitals to be more efficient, influence diagnostic and treatment processes, and to map the spread of a pandemic.

One of the best examples of data scientists making a meaningful difference at a global level is in the response to the COVID-19 pandemic, where they have improved information collection, provided ongoing and accurate estimates of infection spread and health system demand, and assessed the effectiveness of government policies.

**The company wants to know:**

- Which variables are significant in predicting the reason for hospitalization for different regions;
- How well some variables like viral load, smoking, and severity level describe the hospitalization charges;


## Data Description

The file apollo_data.csv contains anonymized data of COVID-19 hospital patients and contains the following variables:

- `age` - an integer indicating the age of the primary beneficiary (excluding those above 64 years, since they are generally covered by the government)
- `sex` - the policy holder's gender, either male or female
- `smoker` - 'yes' or 'no' depending on whether the insured regularly smokes tobacco
- `region` - beneficiary's place of residence in Delhi, divided into four geographic regions - northeast, southeast, southwest, or northwest
- `viral load` - the amount of virus in an infected person's blood
- `severity level` - an integer indicating how severe the patient is
- `hospitalization charges` - individual medical costs billed to health insurance. The currency for the charge has not been set so we will assume it to be USD(\\$)

Import necessary libraries.

In [1]:
import pandas as pd
import numpy as np

## Data Exploration

In [2]:
df = pd.read_csv('apollo_data.csv')
df.head()

Unnamed: 0  age     sex smoker     region  viral load  severity level  \
0           0   19  female    yes  southwest        9.30               0   
1           1   18    male     no  southeast       11.26               1   
2           2   28    male     no  southeast       11.00               3   
3           3   33    male     no  northwest        7.57               0   
4           4   32    male     no  northwest        9.63               0   

   hospitalization charges  
0                    42212  
1                     4314  
2                    11124  
3                    54961  
4                     9667

We drop the unnecessary column named 'Unnamed: 0'

In [3]:
df.drop(columns=['Unnamed: 0'], inplace=True)
df.head()

age     sex smoker     region  viral load  severity level  \
0   19  female    yes  southwest        9.30               0   
1   18    male     no  southeast       11.26               1   
2   28    male     no  southeast       11.00               3   
3   33    male     no  northwest        7.57               0   
4   32    male     no  northwest        9.63               0   

   hospitalization charges  
0                    42212  
1                     4314  
2                    11124  
3                    54961  
4                     9667

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   age                      1338 non-null   int64  
 1   sex                      1338 non-null   object 
 2   smoker                   1338 non-null   object 
 3   region                   1338 non-null   object 
 4   viral load               1338 non-null   float64
 5   severity level           1338 non-null   int64  
 6   hospitalization charges  1338 non-null   int64  
dtypes: float64(1), int64(3), object(3)
memory usage: 73.3+ KB


In [5]:
df.describe()

age   viral load  severity level  hospitalization charges
count  1338.000000  1338.000000     1338.000000              1338.000000
mean     39.207025    10.221233        1.094918             33176.058296
std      14.049960     2.032796        1.205493             30275.029296
min      18.000000     5.320000        0.000000              2805.000000
25%      27.000000     8.762500        0.000000             11851.000000
50%      39.000000    10.130000        1.000000             23455.000000
75%      51.000000    11.567500        2.000000             41599.500000
max      64.000000    17.710000        5.000000            159426.000000

There is no missing value.

In [6]:
df.isna().sum()

age                        0
sex                        0
smoker                     0
region                     0
viral load                 0
severity level             0
hospitalization charges    0
dtype: int64

## Business Requirements

We will go through the business requirements:
- *Which variables are significant in predicting the reason for hospitalization for different regions*
- *How well some variables like viral load, smoking, and severity level describe the hospitalization charges*

We first encode the variables with dtype object by getting the dummies of the categorical variables.

In [7]:
df['sex'].unique()

array(['female', 'male'], dtype=object)

In [8]:
df['smoker'].unique()

array(['yes', 'no'], dtype=object)

In [9]:
df['region'].unique()

array(['southwest', 'southeast', 'northwest', 'northeast'], dtype=object)

In [10]:
df_encoded = pd.get_dummies(df, columns=['sex', 'smoker', 'region'])
df_encoded.head()

age  viral load  severity level  hospitalization charges  sex_female  \
0   19        9.30               0                    42212        True   
1   18       11.26               1                     4314       False   
2   28       11.00               3                    11124       False   
3   33        7.57               0                    54961       False   
4   32        9.63               0                     9667       False   

   sex_male  smoker_no  smoker_yes  region_northeast  region_northwest  \
0     False      False        True             False             False   
1      True       True       False             False             False   
2      True       True       False             False             False   
3      True       True       False             False              True   
4      True       True       False             False              True   

   region_southeast  region_southwest  
0             False              True  
1              True             False  
2              True             False  
3             False             False  
4             False             False

Since the encoded values are in boolean we will change it to be 0 and 1.

In [11]:
df_encoded.columns

Index(['age', 'viral load', 'severity level', 'hospitalization charges',
       'sex_female', 'sex_male', 'smoker_no', 'smoker_yes', 'region_northeast',
       'region_northwest', 'region_southeast', 'region_southwest'],
      dtype='object')

In [12]:
import statsmodels.api as sm

boolean_cols = ['sex_female', 'sex_male', 'smoker_no', 'smoker_yes', 'region_northeast',
       'region_northwest', 'region_southeast', 'region_southwest']

for col in boolean_cols:
    df_encoded[col] = df_encoded[col].astype(int)
    
df_encoded.head()

age  viral load  severity level  hospitalization charges  sex_female  \
0   19        9.30               0                    42212           1   
1   18       11.26               1                     4314           0   
2   28       11.00               3                    11124           0   
3   33        7.57               0                    54961           0   
4   32        9.63               0                     9667           0   

   sex_male  smoker_no  smoker_yes  region_northeast  region_northwest  \
0         0          0           1                 0                 0   
1         1          1           0                 0                 0   
2         1          1           0                 0                 0   
3         1          1           0                 0                 1   
4         1          1           0                 0                 1   

   region_southeast  region_southwest  
0                 0                 1  
1                 1                 0  
2                 1                 0  
3                 0                 0  
4                 0                 0

Now let's fit it to the OLS(Ordinary Least Squares) Regression model.

In [13]:
X = sm.add_constant(df_encoded[['age', 'viral load', 'severity level',
       'sex_female', 'sex_male', 'smoker_no', 'smoker_yes', 'region_northeast',
       'region_northwest', 'region_southeast', 'region_southwest']])
y = df_encoded['hospitalization charges']

model = sm.OLS(y, X).fit()
print(model.summary())

                               OLS Regression Results                              
Dep. Variable:     hospitalization charges   R-squared:                       0.751
Model:                                 OLS   Adj. R-squared:                  0.749
Method:                      Least Squares   F-statistic:                     500.9
Date:                     Tue, 19 Nov 2024   Prob (F-statistic):               0.00
Time:                             13:46:18   Log-Likelihood:                -14774.
No. Observations:                     1338   AIC:                         2.957e+04
Df Residuals:                         1329   BIC:                         2.961e+04
Df Model:                                8                                         
Covariance Type:                 nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------

**1. Model Summary**

- R-squared: 0.751 means 75.1% of the variance in hospitalization charges is explained by the model.
- F-statistic: 500.9 with the P-value converging to 0. 

**2. Coefficient**

- Higher absolute coefficient means a stronger significance level to the target variable. 
- The patient being a smoker or not has the strongest significance in deciding the hospitalization charges. `smoker_no` has a coef of -30180, meaning if the patient is a non smoker the charge of the hospitalization will drop by \\$30180
- Being a smoker will increase the charge to \\$29440

**3. Statistical Significance**

- `sex` variables both have a high p-value more than the significance level 0.05; `sex_female` 0.758, `sex_male` 0.436 meaning the variables are not significantly correlated to the target variable. 
- `region` also seems to have a lower significance; all region related variable have p-value level higher than 0.05. 
- `smoker` variables both have the p-value converging to 0 which is smaller than 0.05. We can say smoker variable is has the highest significance in predicting the reason for hospitalization charge. 
- `viral load` and `severity level` also have a strong significance to the target variable. Also the p-value for both variables are less than 0.05.

## Divide by Region

Since our first business requirement is asking to check which variables are significant in predicting the reason for hospitalization for different regions, we will separate the `df_encoded` to 4 regions. Then fit in the regression model. We know that region is not a significant variable for predicting the hospitalization charges from above result but still will check it in case. 

In [21]:
df_northeast = df_encoded[df_encoded['region_northeast']==True]
df_northeast

age  viral load  severity level  hospitalization charges  sex_female  \
8      37        9.94               2                    16016           0   
10     25        8.74               0                     6803           0   
16     52       10.26               1                    26993           1   
17     23        7.95               0                     5988           0   
20     60       12.00               0                    33072           1   
...   ...         ...             ...                      ...         ...   
1321   62        8.90               0                    70253           0   
1325   61       11.18               0                    32858           0   
1326   42       10.96               0                    17625           1   
1328   23        8.08               2                    55989           1   
1334   18       10.64               0                     5515           1   

      sex_male  smoker_no  smoker_yes  region_northeast  region_northwest  \
8            1          1           0                 1                 0   
10           1          1           0                 1                 0   
16           0          1           0                 1                 0   
17           1          1           0                 1                 0   
20           0          1           0                 1                 0   
...        ...        ...         ...               ...               ...   
1321         1          0           1                 1                 0   
1325         1          1           0                 1                 0   
1326         0          1           0                 1                 0   
1328         0          1           0                 1                 0   
1334         0          1           0                 1                 0   

      region_southeast  region_southwest  
8                    0                 0  
10                   0                 0  
16                   0                 0  
17                   0                 0  
20                   0                 0  
...                ...               ...  
1321                 0                 0  
1325                 0                 0  
1326                 0                 0  
1328                 0                 0  
1334                 0                 0  

[324 rows x 12 columns]

In [30]:
X = sm.add_constant(df_northeast[['age', 'viral load', 'severity level',
       'sex_female', 'sex_male', 'smoker_no', 'smoker_yes']])
y = df_northeast['hospitalization charges']

model = sm.OLS(y, X).fit()
print(model.summary())

                               OLS Regression Results                              
Dep. Variable:     hospitalization charges   R-squared:                       0.709
Model:                                 OLS   Adj. R-squared:                  0.705
Method:                      Least Squares   F-statistic:                     155.2
Date:                     Tue, 19 Nov 2024   Prob (F-statistic):           4.44e-83
Time:                             14:52:26   Log-Likelihood:                -3578.4
No. Observations:                      324   AIC:                             7169.
Df Residuals:                          318   BIC:                             7192.
Df Model:                                5                                         
Covariance Type:                 nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------

In [18]:
df_northwest = df_encoded[df_encoded['region_northwest']==True]

In [31]:
X = sm.add_constant(df_northwest[['age', 'viral load', 'severity level',
       'sex_female', 'sex_male', 'smoker_no', 'smoker_yes']])
y = df_northwest['hospitalization charges']

model = sm.OLS(y, X).fit()
print(model.summary())

                               OLS Regression Results                              
Dep. Variable:     hospitalization charges   R-squared:                       0.704
Model:                                 OLS   Adj. R-squared:                  0.700
Method:                      Least Squares   F-statistic:                     152.0
Date:                     Tue, 19 Nov 2024   Prob (F-statistic):           3.45e-82
Time:                             14:52:33   Log-Likelihood:                -3586.9
No. Observations:                      325   AIC:                             7186.
Df Residuals:                          319   BIC:                             7208.
Df Model:                                5                                         
Covariance Type:                 nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------

In [19]:
df_southeast = df_encoded[df_encoded['region_southeast']==True]

In [32]:
X = sm.add_constant(df_southeast[['age', 'viral load', 'severity level',
       'sex_female', 'sex_male', 'smoker_no', 'smoker_yes']])
y = df_southeast['hospitalization charges']

model = sm.OLS(y, X).fit()
print(model.summary())

                               OLS Regression Results                              
Dep. Variable:     hospitalization charges   R-squared:                       0.799
Model:                                 OLS   Adj. R-squared:                  0.796
Method:                      Least Squares   F-statistic:                     284.8
Date:                     Tue, 19 Nov 2024   Prob (F-statistic):          2.17e-122
Time:                             14:52:38   Log-Likelihood:                -4031.7
No. Observations:                      364   AIC:                             8075.
Df Residuals:                          358   BIC:                             8099.
Df Model:                                5                                         
Covariance Type:                 nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------

In [20]:
df_southwest = df_encoded[df_encoded['region_southwest']==True]

In [33]:
X = sm.add_constant(df_southwest[['age', 'viral load', 'severity level',
       'sex_female', 'sex_male', 'smoker_no', 'smoker_yes']])
y = df_southwest['hospitalization charges']

model = sm.OLS(y, X).fit()
print(model.summary())

                               OLS Regression Results                              
Dep. Variable:     hospitalization charges   R-squared:                       0.786
Model:                                 OLS   Adj. R-squared:                  0.783
Method:                      Least Squares   F-statistic:                     234.3
Date:                     Tue, 19 Nov 2024   Prob (F-statistic):          1.75e-104
Time:                             14:52:42   Log-Likelihood:                -3548.3
No. Observations:                      325   AIC:                             7109.
Df Residuals:                          319   BIC:                             7131.
Df Model:                                5                                         
Covariance Type:                 nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------

Even if we separate the dataframe according to the regions, the tendency of the regression models follow a similar path; `smoker` variable being the highest correlation to the target variable `hospitalization charges`, and `sex` having a high p-value meaning it doesn't have a strong correlation to the target variable. 

## Conclusion

In [34]:
X = sm.add_constant(df_encoded[['age', 'viral load', 'severity level',
       'sex_female', 'sex_male', 'smoker_no', 'smoker_yes', 'region_northeast',
       'region_northwest', 'region_southeast', 'region_southwest']])
y = df_encoded['hospitalization charges']

model = sm.OLS(y, X).fit()
print(model.summary())

                               OLS Regression Results                              
Dep. Variable:     hospitalization charges   R-squared:                       0.751
Model:                                 OLS   Adj. R-squared:                  0.749
Method:                      Least Squares   F-statistic:                     500.9
Date:                     Tue, 19 Nov 2024   Prob (F-statistic):               0.00
Time:                             17:16:17   Log-Likelihood:                -14774.
No. Observations:                     1338   AIC:                         2.957e+04
Df Residuals:                         1329   BIC:                         2.961e+04
Df Model:                                8                                         
Covariance Type:                 nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------

To answer our first business requirement, `smoker` variable has the highest significance in predicting the reason for hospitalization charges for different regions.

We can also answer the second business requirement; How well some variables like viral load, smoking, and severity level describe the hospitalization charges

- `viral load`, `smoker` and `severity level` all has p-value less than the significance level meaning they are highly correlated to predicting the target variable.

**smoker variable**
- The patient being a smoker or not has the strongest significance in deciding the hospitalization charges. `smoker_no` has a coef of -30180, meaning if the patient is a non-smoker the charge of the hospitalization will minus by \\$30180
- Being a smoker will increase the charge to \\$29440
- According to the [0.025 and 0.975] we can check the 95% confidence interval for each coef. For the `smoker_no` variable we can be 95% confident that the true effect for a patient being non-smoker could reduce the hospitalization charge from \\$31600 to \\$28800.
- Same goes with the `smoker_yes` variable; 95% confidence for the patient being a smoker could increase the hospitalization charge from \\$27900 to \\$31000.

**viral load**
- `viral load` is the third variable that is strongly correlated to when it comes to deciding the hospitalization charges. It has a coef of 2544, meaning when each unit increases in viral load, charges increase by about \\$2544.
- The amount could be up from \\$2123 to \\$2965. 

**severity level**
- `severity level` follows next after `viral load`. With a coef of 1188, the hospitalization charge increases by a unit of increase on severity level by around \\$1188.
- The amount could be up from \\$513 to \\$1864.